### PR TITLE
re-enabling testing to see failures

### PR DESCRIPTION
**Pull request by larkost**
Will address #4472.

### CODE REVIEW

**Comment by larkost** [comment]
Retest this please

**Comment by larkost** [comment]
I think that this solves the issue at hand (`iOS-swift-device` failing to build), and it seems that the cause was a missing dependency on the `RealmSwift Tests`. It is not immediately apparent why this worked somewhere in the past, but it might be that Tests units listed were implicitly built in a previous version of Xcode, but now need to be explicit dependencies.

I should also note that in the schema editor it appeared that this would be built. So this might well be a new(-ish) bug in Xcode.

The tests are failing in this run, but they are being run:
https://ci.realm.io/job/objc_pr/configuration=Release,swift_version=3.0.2,target=ios-device-swift/4492/

File: build.sh [hunk 1]
```diff
@@ -335,12 +335,13 @@ download_sync() {
     TMP_DIR="$TMPDIR/sync_bin"
     mkdir -p "${TMP_DIR}"
     SYNC_TMP_TAR="${TMP_DIR}/sync-${REALM_SYNC_VERSION}.tar.xz.tmp"
+    rm -rf "$SYNC_TMP_TAR" # ensure we don't have a corrupt download
```
Comment (jpsim) [inline build.sh]: not much point in caching the download if we just delete the tarball on every run...

if this is to handle corrupt downloads, it should be done as error handling rather than preemptively.

File: build.sh [hunk 2]
```diff
@@ -335,12 +335,13 @@ download_sync() {
...
     SYNC_TAR="${TMP_DIR}/sync-${REALM_SYNC_VERSION}.tar.xz"
     if [ ! -f "${SYNC_TAR}" ]; then
         local SYNC_URL="https://static.realm.io/downloads/sync/realm-sync-cocoa-${REALM_SYNC_VERSION}.tar.xz"
         set +e # temporarily disable immediate exit
         local ERROR # sweeps the exit code unless declared separately
-        ERROR=$(curl --fail --silent --show-error --location "$SYNC_URL" --output "${SYNC_TMP_TAR}" 2>&1 >/dev/null)
+        ERROR=$(curl --fail --silent --show-error --location "$SYNC_URL" --output "${SYNC_TMP_TAR}" 2>&1)
```
Comment (jpsim) [inline build.sh]: why this change? We typically want to avoid polluting our users' build logs with curl output.

File: build.sh (hunk 1 above)
Comment (larkost) [inline build.sh]: This isn't a change that adds caching, it is a change to delete a potentially bad version that is already there. So we already had it "cached", this just avoids a problem I ran into during testing.

Comment (jpsim) [inline build.sh]: I know this doesn't add caching. It removes the caching feature that's been there for years.

What I was saying is if you want to preemptively delete the archive regardless of whether or not it is valid or corrupt, it breaks caching entirely.

If this is to handle corrupt downloads, it should be done as error handling rather than preemptively.

File: build.sh (hunk 2 above)
Comment (larkost) [inline build.sh]: The output is already being captured into the `$ERROR` variable, and only displayed when there is a problem. Before this change we were tossing everything away, making troubleshooting unnecessarily hard.

Comment (jpsim) [inline build.sh]: I see. Sounds good then.

File: build.sh (hunk 1 above)
Comment (larkost) [inline build.sh]: Looking at it more I don't think that we should make this that much more complicated. The file that is being deleted is `$SYNC_TMP_TAR`, which if it is successfully downloaded gets moved to `$SYNC_TAR`. So the only files that could be at `$SYNC_TMP_TAR` to be deleted would be a failed (or theoretically in-progress) file. There is room for improvement in this area, but `$SYNC_TMP_TAR`, and some implicit caching is nice, but it does not seem the place for that.

Comment (jpsim) [inline build.sh]: Oh, so great! I hadn't noticed (and forgot) the distinction between `$SYNC_TMP_TAR` and `$SYNC_TAR`.

**Comment by jpsim** [comment]
Given that the main purpose of this PR is to fix iOS Swift device CI job, could you ping me again when that job is passing CI?

Appears to be failing: https://ci.realm.io/job/objc_pr/4493/configuration=Release,swift_version=3.0.2,target=ios-device-swift/console

**Comment by larkost** [comment]
I just rebased off of `master` to re-trigger a check.

**Comment by larkost** [comment]
In the latest run we are back to just having what look like valid test failures in [`ios-device-swift`](https://ci.realm.io/job/objc_pr/configuration=Release,swift_version=3.0.2,target=ios-device-swift/4528/console) and [`ios-device-objc`](https://ci.realm.io/job/objc_pr/configuration=Release,swift_version=3.0.2,target=ios-device-objc/4529/console). So I think we are good.

**Comment by jpsim** [comment]
The git operations went awry in this branch.

image

@larkost please fix

**Comment by jpsim** [comment]
I believe _some_ of the failures are due to the bad git operations in this branch. For example, for the ObjC device tests:

```
2017-01-04 19:15:31.039 TestHost[2857:3836966] +[NSExpression expressionForConditional:trueExpression:falseExpression:]: unrecognized selector sent to class 0x3acce7f4
<unknown>:0: error: -[PredicateUtilTests testVisitingAllExpressionTypes] : failed: caught "NSInvalidArgumentException", "+[NSExpression expressionForConditional:trueExpression:falseExpression:]: unrecognized selector sent to class 0x3acce7f4"
(
	0   CoreFoundation                      0x2ca264b7 <redacted> + 150
	1   libobjc.A.dylib                     0x3a1e0c8b objc_exception_throw + 38
	2   CoreFoundation                      0x2ca2b7d5 <redacted> + 0
	3   CoreFoundation                      0x2ca297d7 <redacted> + 714
	4   CoreFoundation                      0x2c95b058 _CF_forwarding_prep_0 + 24
	5   Realm                               0x00928727 _ZNK12_GLOBAL__N_130PredicateExpressionTransformer5visitEP12NSExpression + 2242
	6   Realm                               0x00927c79 _ZNK12_GLOBAL__N_130PredicateExpressionTransformer5visitEP11NSPredicate + 672
	7   Realm                               0x009278f7 _Z18transformPredicateP11NSPredicateNSt3__18functionIFP12NSExpressionS4_EEE + 186
	8   Tests                               0x002fc24d _ZZ52-[PredicateUtilTests testVisitingAllExpressionTypes]ENK3$_0clEP11NSPredicatem + 144
	9   Tests                               0x002fbef9 -[PredicateUtilTests testVisitingAllExpressionTypes] + 752
	10  CoreFoundation                      0x2ca2ab14 <redacted> + 68
	11  CoreFoundation                      0x2c957695 <redacted> + 300
	12  XCTest                              0x001e5b55 __24-[XCTestCase invokeTest]_block_invoke_2 + 472
	13  XCTest                              0x00218efd -[XCTestContext performInScope:] + 312
	14  XCTest                              0x001e5977 -[XCTestCase invokeTest] + 296
	15  Tests                               0x006f538f -[RLMTestCase invokeTest] + 310
	16  XCTest                              0x001e61a9 -[XCTestCase performTest:] + 560
	17  XCTest                              0x001e34fd -[XCTestSuite performTest:] + 520
	18  XCTest                              0x001e34fd -[XCTestSuite performTest:] + 520
	19  XCTest                              0x001e34fd -[XCTestSuite performTest:] + 520
	20  XCTest                              0x001cf805 __25-[XCTestDriver _runSuite]_block_invoke + 52
	21  XCTest                              0x001f0095 -[XCTestObservationCenter _observeTestExecutionForBlock:] + 628
	22  XCTest                              0x001cf69d -[XCTestDriver _runSuite] + 512
	23  XCTest                              0x001d04d3 -[XCTestDriver _checkForTestManager] + 300
	24  XCTest                              0x0021a37b _XCTestMain + 690
	25  CoreFoundation                      0x2c9ecd7d <redacted> + 12
	26  CoreFoundation                      0x2c9ec041 <redacted> + 216
	27  CoreFoundation                      0x2c9ea7c3 <redacted> + 762
	28  CoreFoundation                      0x2c9383c1 CFRunLoopRunSpecific + 476
	29  CoreFoundation                      0x2c9381d3 CFRunLoopRunInMode + 106
	30  GraphicsServices                    0x33d360a9 GSEventRunModal + 136
	31  UIKit                               0x2ff467b1 UIApplicationMain + 1440
	32  TestHost                            0x000905b9 main + 70
	33  libdyld.dylib                       0x3a760aaf <redacted> + 2
)
Test Case '-[PredicateUtilTests testVisitingAllExpressionTypes]' failed (0.162 seconds).
```

Looking at the diff between this branch and `master`, it looks like this branch has messed up that code.

And the Swift device test job failed for Jenkins-related reasons:

```
ERROR: Build step failed with exception
java.lang.IllegalStateException: Invalid object ID 5406 iota=5407
	at hudson.remoting.ExportTable.diagnoseInvalidId(ExportTable.java:386)
	at hudson.remoting.ExportTable.get(ExportTable.java:330)
	at hudson.remoting.Channel.getExportedObject(Channel.java:633)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.perform(RemoteInvocationHandler.java:875)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:859)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:818)
	at hudson.remoting.UserRequest.perform(UserRequest.java:153)
	at hudson.remoting.UserRequest.perform(UserRequest.java:50)
	at hudson.remoting.Request$2.run(Request.java:332)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at hudson.remoting.Engine$1$1.run(Engine.java:85)
	at java.lang.Thread.run(Thread.java:745)
	at ......remote call to sf_host1_1(Native Method)
	at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1433)
	at hudson.remoting.UserResponse.retrieve(UserRequest.java:253)
	at hudson.remoting.Channel.call(Channel.java:797)
	at hudson.remoting.RemoteInvocationHandler.invoke(RemoteInvocationHandler.java:252)
	at com.sun.proxy.$Proxy90.join(Unknown Source)
	at hudson.Launcher$RemoteLauncher$ProcImpl.join(Launcher.java:991)
	at hudson.tasks.CommandInterpreter.join(CommandInterpreter.java:135)
	at hudson.tasks.CommandInterpreter.perform(CommandInterpreter.java:95)
	at hudson.tasks.CommandInterpreter.perform(CommandInterpreter.java:64)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.Build$BuildExecution.build(Build.java:205)
	at hudson.model.Build$BuildExecution.doRun(Build.java:162)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:534)
	at hudson.model.Run.execute(Run.java:1720)
	at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:404)
Caused by: java.lang.Exception: Object was recently deallocated
    #5406 (ref.0) : object=null type=hudson.Launcher$RemoteLaunchCallable$1 interfaces=[hudson.Launcher$RemoteProcess]
      Created at Wed Jan 04 19:10:10 PST 2017
    	at hudson.remoting.ExportTable$Entry.<init>(ExportTable.java:99)
    	at hudson.remoting.ExportTable.export(ExportTable.java:305)
    	at hudson.remoting.Channel.internalExport(Channel.java:629)
    	at hudson.remoting.Channel.export(Channel.java:620)
    	at hudson.remoting.Channel.export(Channel.java:590)
    	at hudson.Launcher$RemoteLaunchCallable.call(Launcher.java:1150)
    	at hudson.Launcher$RemoteLaunchCallable.call(Launcher.java:1113)
    	at hudson.remoting.UserRequest.perform(UserRequest.java:153)
    	at hudson.remoting.UserRequest.perform(UserRequest.java:50)
    	at hudson.remoting.Request$2.run(Request.java:332)
    	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
    	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    	at hudson.remoting.Engine$1$1.run(Engine.java:85)
    	at java.lang.Thread.run(Thread.java:745)
      Released at Wed Jan 04 19:10:11 PST 2017
    	at hudson.remoting.ExportTable$Entry.release(ExportTable.java:131)
    	at hudson.remoting.ExportTable.unexportByOid(ExportTable.java:414)
    	at hudson.remoting.Channel.unexport(Channel.java:641)
    	at hudson.remoting.UnexportCommand.execute(UnexportCommand.java:43)
    	at hudson.remoting.Channel$1.handle(Channel.java:501)
    	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:83)
    Caused by: Command hudson.remoting.UnexportCommand@50429d3a created at
    	at hudson.remoting.Command.<init>(Command.java:68)
    	at hudson.remoting.Command.<init>(Command.java:51)
    	at hudson.remoting.UnexportCommand.<init>(UnexportCommand.java:33)
    	at hudson.remoting.RemoteInvocationHandler$PhantomReferenceImpl.cleanup(RemoteInvocationHandler.java:362)
    	at hudson.remoting.RemoteInvocationHandler$PhantomReferenceImpl.access$900(RemoteInvocationHandler.java:321)
    	at hudson.remoting.RemoteInvocationHandler$Unexporter.run(RemoteInvocationHandler.java:579)
    	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    	at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:110)
    	at java.lang.Thread.run(Thread.java:745)
    Caused by: java.lang.Exception: Proxy hudson.remoting.RemoteInvocationHandler@151e was created for interface hudson.Launcher$RemoteProcess
    	at hudson.remoting.RemoteInvocationHandler.<init>(RemoteInvocationHandler.java:124)
    	at hudson.remoting.RemoteInvocationHandler.wrap(RemoteInvocationHandler.java:136)
    	at hudson.remoting.Channel.export(Channel.java:621)
    	at hudson.remoting.Channel.export(Channel.java:590)
    	at hudson.Launcher$RemoteLaunchCallable.call(Launcher.java:1150)
    	at hudson.Launcher$RemoteLaunchCallable.call(Launcher.java:1113)
    	at hudson.remoting.UserRequest.perform(UserRequest.java:153)
    	at hudson.remoting.UserRequest.perform(UserRequest.java:50)
    	at hudson.remoting.Request$2.run(Request.java:332)
    	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
    	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    	at hudson.remoting.Engine$1$1.run(Engine.java:85)
    	... 1 more
	at hudson.remoting.ExportTable.diagnoseInvalidId(ExportTable.java:379)
	at hudson.remoting.ExportTable.get(ExportTable.java:330)
	at hudson.remoting.Channel.getExportedObject(Channel.java:633)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.perform(RemoteInvocationHandler.java:875)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:859)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:818)
	at hudson.remoting.UserRequest.perform(UserRequest.java:153)
	at hudson.remoting.UserRequest.perform(UserRequest.java:50)
	at hudson.remoting.Request$2.run(Request.java:332)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at hudson.remoting.Engine$1$1.run(Engine.java:85)
	at java.lang.Thread.run(Thread.java:745)
Caused by:   Released at Thu Jan 05 03:10:11 UTC 2017
	at hudson.remoting.ExportTable$Entry.release(ExportTable.java:131)
	at hudson.remoting.ExportTable.unexportByOid(ExportTable.java:414)
	at hudson.remoting.Channel.unexport(Channel.java:641)
	at hudson.remoting.UnexportCommand.execute(UnexportCommand.java:43)
	at hudson.remoting.Channel$1.handle(Channel.java:501)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:83)
Caused by: Command hudson.remoting.UnexportCommand@274331dc created at
	at hudson.remoting.Command.<init>(Command.java:68)
	at hudson.remoting.Command.<init>(Command.java:51)
	at hudson.remoting.UnexportCommand.<init>(UnexportCommand.java:33)
	at hudson.remoting.RemoteInvocationHandler$PhantomReferenceImpl.cleanup(RemoteInvocationHandler.java:362)
	at hudson.remoting.RemoteInvocationHandler$PhantomReferenceImpl.access$900(RemoteInvocationHandler.java:321)
	at hudson.remoting.RemoteInvocationHandler$Unexporter.run(RemoteInvocationHandler.java:579)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:110)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.Exception: Proxy hudson.remoting.RemoteInvocationHandler@151e was created for interface hudson.Launcher$RemoteProcess
	at hudson.remoting.RemoteInvocationHandler.<init>(RemoteInvocationHandler.java:124)
	at hudson.remoting.RemoteInvocationHandler.wrap(RemoteInvocationHandler.java:136)
	at hudson.remoting.Channel.export(Channel.java:621)
	at hudson.remoting.Channel.export(Channel.java:590)
	at hudson.Launcher$RemoteLaunchCallable.call(Launcher.java:1150)
	at hudson.Launcher$RemoteLaunchCallable.call(Launcher.java:1113)
	at hudson.remoting.UserRequest.perform(UserRequest.java:153)
	at hudson.remoting.UserRequest.perform(UserRequest.java:50)
	at hudson.remoting.Request$2.run(Request.java:332)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at hudson.remoting.Engine$1$1.run(Engine.java:85)
	... 1 more
Build step 'Execute shell' marked build as failure
/Users/realm/.gems/gems/json-1.8.3/lib/json/common.rb:155:in `initialize': A JSON text must at least contain two octets! (JSON::ParserError)
	from /Users/realm/.gems/gems/json-1.8.3/lib/json/common.rb:155:in `new'
	from /Users/realm/.gems/gems/json-1.8.3/lib/json/common.rb:155:in `parse'
	from -e:1:in `<main>'
Adding comment...
```

---

In general, we won't be able to merge this until tests are not only _running_ reliably, but also _passing_ reliably. So work on this won't be done until that point is reached.

**Comment by larkost** [comment]
@jpsim I think you are being confused by GitHub not handling rebases well for their difftool. If you look at the changes from `master` like this (note `origin/master` was at 7649d311 at the time I did the rebase):
`git difftool -d origin/master origin/larkost/4472-reenable-swift-device-testing`
... then you can see only the changes that this would actually bring. Additionally, if you look at the links to the results I provided, you can see that both of them are failing unit tests. Really failing them, specifically:
`PredicateUtilTests
    ✗ testVisitingAllExpressionTypes, failed: caught "NSInvalidArgumentException", "+[NSExpression expressionForConditional:trueExpression:falseExpression:]: unrecognized selector sent to class 0x3acce7f4"`
for `ios-device-objc`, and 14 tests for `ios-device-swift` all because of time regressions:
`
SwiftPerformanceTests
    ◷ testCommitWriteTransaction measured (0.025 seconds)
    ✓ testCommitWriteTransaction (1.756 seconds)
    ◷ testCommitWriteTransactionWithCrossThreadNotification measured (0.039 seconds)
    ✓ testCommitWriteTransactionWithCrossThreadNotification (1.888 seconds)
    ◷ testCommitWriteTransactionWithLocalNotification measured (0.027 seconds)
    ✓ testCommitWriteTransactionWithLocalNotification (1.907 seconds)
    ◷ testCountWhereQuery measured (0.291 seconds)
    ✓ testCountWhereQuery (3.239 seconds)
    ◷ testCountWhereTableView measured (0.143 seconds)
    ✓ testCountWhereTableView (1.732 seconds)
    ◷ testCrossThreadSyncLatency measured (0.213 seconds)
    ✓ testCrossThreadSyncLatency (3.772 seconds)
    ◷ testDeleteAll measured (0.062 seconds)
    ✗ testDeleteAll, failed: Time average is 18% worse (max allowed: 10%).
    ◷ testEnumerateAndAccessAll measured (1.703 seconds)
    ✗ testEnumerateAndAccessAll, failed: Time average is 24% worse (max allowed: 10%).
    ◷ testEnumerateAndAccessAllSlow measured (1.580 seconds)
    ✗ testEnumerateAndAccessAllSlow, failed: Time average is 12% worse (max allowed: 10%).
    ◷ testEnumerateAndAccessArrayProperty measured (1.738 seconds)
    ✗ testEnumerateAndAccessArrayProperty, failed: Time average is 26% worse (max allowed: 10%).
    ◷ testEnumerateAndAccessArrayPropertySlow measured (1.704 seconds)
    ✗ testEnumerateAndAccessArrayPropertySlow, failed: Time average is 11% worse (max allowed: 10%).
    ◷ testEnumerateAndAccessQuery measured (0.990 seconds)
    ✗ testEnumerateAndAccessQuery, failed: Time average is 29% worse (max allowed: 10%).
    ◷ testEnumerateAndMutateAll measured (2.809 seconds)
    ✗ testEnumerateAndMutateAll, failed: Time average is 42% worse (max allowed: 10%).
    ◷ testEnumerateAndMutateQuery measured (1.448 seconds)
    ✗ testEnumerateAndMutateQuery, failed: Time average is 39% worse (max allowed: 10%).
    ◷ testIndexedStringLookup measured (0.247 seconds)
    ✗ testIndexedStringLookup, failed: Time average is 16% worse (max allowed: 10%).
    ◷ testInsertMultiple measured (0.272 seconds)
    ✗ testInsertMultiple, failed: Time average is 13% worse (max allowed: 10%).
    ◷ testInsertMultipleLiteral measured (0.405 seconds)
    ✗ testInsertMultipleLiteral, failed: Time average is 24% worse (max allowed: 10%).
    ◷ testInsertSingleLiteral measured (0.128 seconds)
    ✓ testInsertSingleLiteral (1.900 seconds)
    ◷ testLargeINQuery measured (0.450 seconds)
    ✓ testLargeINQuery (5.795 seconds)
    ◷ testManualDeletion measured (0.823 seconds)
    ✗ testManualDeletion, failed: Time average is 69% worse (max allowed: 10%).
    ◷ testQueryDeletion measured (0.271 seconds)
    ✗ testQueryDeletion, failed: Time average is 136% worse (max allowed: 10%).
    ◷ testRealmCreationCached measured (0.451 seconds)
    ✓ testRealmCreationCached (4.936 seconds)
    ◷ testRealmCreationUncached measured (0.577 seconds)
    ✓ testRealmCreationUncached (6.041 seconds)
    ◷ testSortingAllObjects measured (0.059 seconds)
    ✓ testSortingAllObjects (1.740 seconds)
    ◷ testUnindexedStringLookup measured (0.240 seconds)
    ✗ testUnindexedStringLookup, failed: Time average is 17% worse (max allowed: 10%).
`
These are obviously not CI failures, and should not be blocking this.

**Comment by bdash** [comment]
@larkost, your branch history definitely looks bogus to me.

**Comment by jpsim** [comment]
Yes, there have been runs in which tests ran but failed, but that's not the latest state.

The latest run of CI is here: https://ci.realm.io/job/objc_pr/4529/, which has two failures (the two device test jobs):

* [ios-device-objc](https://ci.realm.io/job/objc_pr/4529/configuration=Release,swift_version=3.0.2,target=ios-device-objc/console)
* [ios-device-swift](https://ci.realm.io/job/objc_pr/4529/configuration=Release,swift_version=3.0.2,target=ios-device-swift/console)

These two failures have the output I shared in my previous comment.

It doesn't much matter what a previous state of the PR was producing when comes time to merge it. The salient information is how it's behaving in its latest state.

---

Also, fixing the infrastructure is not sufficient to merge a PR. The tests also have to be made to pass. Preferably by fixing failing tests, but in this case, I'm willing to entertain simply disabling them given how long we've gone without device tests at all.

**Comment by jpsim** [comment]
@larkost what would you like to do with this?

**Comment by jpsim** [comment]
Closing due to inactivity.
